packages/user-group-service/src/users/typedef.graphql

```diff
@@ -108,16 +157,21 @@ type Mutation {
   addUserFromLDAP(uid: String!): UserType
 }
 
+
 type Query {
   # Fetches specific User by uid, rhatUUID, apiRole, name
   getUsersBy(
     uid: String
     rhatUUID: String
     apiRole: String
     name: String
-  ): [UserType]
+  ): [UserType] @deprecated(reason: "This query is no longer maintained")
   # Fetches all Users
-  listUsers: [UserType]
+  listUsers: [UserType] @deprecated(reason: "This query is no longer maintained")
   # Fetch LDAP Group Members
-  getGroupMembers(cn: String): [LdapType]
+  getGroupMembers(cn: String): [LdapType] @deprecated(reason: "This query is no longer maintained")
+  # Fetches specific User by uid, rhatUuid
+  user(uid: String, rhatUUID: String): RoverUserType
+  # Fetches Users based on criteria
+  users(ldapfield: ldapFieldType, value: String): [RoverUserType]
```

Comment:
```suggestion
  users(ldapfield: ldapFieldType, value: String): [RoverUserType]
```

If the ```users``` query is called for every record resolution it will throw an API call individually. if the response payload is limited this s fine. if it has more payload we need to use javascript promises to resolve within the resolver. this will improve the response.

Reply:
`users` returns the list of users for general criteria like `cn` hence we can anticipate a lower size payload & that too if those specific fields are requested. Also as discussed with @deshmukhmayur the existing queries are not requesting these fields.